### PR TITLE
fix: non number inputs are skipped for height and reach fields during onboarding on Safari

### PR DESCRIPTION
**Pull request by nathantew14**
## Describe your changes
Issue is that the `type = "number"` prop of input fields does not correctly prevent non-numbers from being entered on Safari, but also prevents the input itself from being passed to the object of values used by `react-hook-form`, as well as mutes the `onChange` method, so the invalid height and reach entries are simply skipped (and cannot otherwise be checked)

To fix this, I moved  the `handleClickDoneButton` method to the `onSubmit` method/prop of `FormProvider`, then changed the "Skip"/"Next" button to invoke the `onSubmit` method, so that the default React's form validation which happens upon submitting a form can be used for each page of the onboarding process (without actually submitting the user info to backend).

## Issue ticket number and link
#109 

## Screenshots (if appropriate):
Screenshot 2022-12-25 at 7 10 46 PM

## Checklist before requesting a review

- [x] I have performed a self-review of my code

Does not seem to affect any behaviour on Chrome, updateUser can still be submitted at the end of the onboarding form, and EditProfile seems to still work.

### CODE REVIEW

File: frontend/src/pages/onboarding/UsernameForm.tsx [hunk 1]
```diff
@@ -25,6 +24,7 @@ export const UsernameForm = () => {
         }}
         required
       />
+      <input hidden type="text"/>
```
Comment (therizhao) [inline frontend/src/pages/onboarding/UsernameForm.tsx:27]: Curious why is there a hidden input here?

**Review by therizhao — CHANGES_REQUESTED** [review]

File: frontend/src/pages/onboarding/UsernameForm.tsx (hunk 1 above)
Comment (therizhao) [reply frontend/src/pages/onboarding/UsernameForm.tsx:27]: ok i saw the other pr thx

**Comment by therizhao** [comment]
Hi @nathantew14  u might want to check the lock file. I think can uncommit the changes in the lock file

**Comment by therizhao** [comment]
otherwise lgtm

**Comment by nathantew14** [comment]
@therizhao after much gnashing of teeth I managed to remove the commit that changed the lock file, but now neither my backend nor frontend can be started with yarn locally

**Review by therizhao — APPROVED** [review]
lgtm!

**Comment by therizhao** [comment]
@all-contributors please add @nathantew14 for code

**Pull request merged**